src/Strategy/Ruby/Gemspec.hs

```diff
@@ -0,0 +1,175 @@
+--
+-- Beyond the above, this parser also will consume and ignore '.freeze' or
+-- '.freeze()' that appears at the end of the string. It is a ruby idiom that
+-- turns a string into an immutable version of itself. It appears in some
+-- gemspec files and is safe to ignore.
+--
+-- An edge-case I ignore here is string interpolation. Ruby allows strings
+-- to contain text like `#{<expr>}`. The `<expr>` is evaluated and inserted
+-- into the string in place of the interpolation text. This parser ignores
+-- these and treats them like regular text.
+rubyString :: Parser Text
+rubyString =
+  mconcat <$> (stringText <* optional freezeMethod)
+  where
+    freezeMethod = string ".freeze" >> optional (string "()")
+    betweenDelim :: Char -> Parser [Text]
+    betweenDelim c =
+      let (d1, d2) = selectDelim c
+          -- only the close delimiter can stop parsing, so only check
+          -- for escaped versions of that.
+          delimEscape = string $ "\\" <> toText d2
+       in between
+            (char d1)
+            (char d2)
+            ( many
+                ( try delimEscape
+                    <|> (toText <$> anySingleBut d2)
+                )
+            )
+    pctQ = optional (choice [char 'q', char 'Q'])
+    arbitraryDelim = try $
+      do (char '%' *> pctQ *> lookAhead anySingle) >>= betweenDelim
+    stringText =
+      (betweenDelim '"')
+        <|> (betweenDelim '\'')
+        <|> try arbitraryDelim
+
+data Assignment a = Assignment
+  { label :: Text
+  , value :: a
+  }
+  deriving (Eq, Show)
+
+-- |Parser for a single assignment statement of the form:
+--
+-- > righthand.side = some-value
+--
+-- whitespace around the '=' is ignored. '<lhs> = ' portion of the assignment
+-- must appear on one line. The right-hand side is taken care of by the argument
+-- parser.
+parseRubyAssignment ::
+  -- | Parser for the right-hand side of an assignment
+  Parser a ->
+  Parser (Assignment a)
+parseRubyAssignment rhs = Assignment <$> (labelP <* lexeme (char '=')) <*> valueP
+  where
+    -- edge-case: assumes that the initial label is on one line
+    labelP = takeWhile1P Nothing (\c -> c /= '=' && not (isSpace c))
+    valueP = rhs
+
+lexeme :: Parser a -> Parser a
+lexeme p = space *> p <* space
+
+-- | Ruby array literals look like this:
+--
+-- > [a, b , c]
+--
+-- This parser is parameterized by a parser `p` which parses the elements
+-- in an array. Whitespace around array elements and the commas between them
+-- are consumed and ignored.
+parseRubyArray :: Parser a -> Parser [a]
+parseRubyArray p = char '[' *> sepBy (lexeme p) (char ',') <* char ']'
+
+-- |Ruby has a special syntax for making an array of strings that looks like
+-- these examples:
+--
+-- > %w(foo bar)
+-- > %W(foo bar baz)
+--
+-- This is interpreted as an array of strings. The 'W' variant also allows
+-- interpolation, but as with 'rubyString' these are treated as ordinary text.
+parseRubyWordsArray :: Parser [Text]
+parseRubyWordsArray =
+  char '%'
```

Comment:
Sorry I didn't notice this the first time through, but array literals can also use the same delimiters as string literals. Yet another case where Ruby took too much inspiration from Perl.

https://en.wikibooks.org/wiki/Ruby_Programming/Syntax/Literals#The_.25_Notation

So these all result in an array of words:

```ruby
a = %w(one two three)
a = %w{one two three}
a = %w#one two three#
a = %w<one two three>
```

The exact same thing happens for the `W` variants, but with a capital W you can put string interpolation inside of the words (which we should ignore like we are ignoring it for strings).

I think this might be worth fixing, but only because we already have the machinery you already built for strings that will make this reasonably easy to do.

Reply:
I have incorporated this into my latest commit.